src/js/Analytics.js

```diff
@@ -0,0 +1,180 @@
+'use strict';
+
+const Analytics = function (trackingId, userId, appName, appVersion, gitRevision, os, checkForDebugVersions, optOut, debugMode, buildType) {
+    this._trackingId = trackingId;
+
+    this.setOptOut(optOut);
+
+    this._googleAnalytics = googleAnalytics;
+
+    this._googleAnalytics.initialize(this._trackingId, {
+        storage: 'none',
+        clientId: userId,
+        debug: !!debugMode,
```

Comment:
Why not esm class here? `prototype` based classes are relic of the past. VScode can refactor that for you.